### DISCUSSION TITLE
How can I get a list of sales orders?

### DISCUSSION THREAD

**Discussion by ataylor32** [netsuitephp/netsuite-php#110] [Q&A] [answered]
Apologies if this is a simple question. I'm very new to both NetSuite and this package.

**Comment by neclimdul** [comment, marked as answer]
There is a "getAll" method but it only supports a small number of types and if I'm reading things correctly sales orders aren't in that. Probably good since always getting them all is the sort of things that would work at launch and fail a month later :)

From here(at the bottom) looks like you'll want to do one of the Transaction search.
https://system.netsuite.com/help/helpcenter/en_US/srbrowser/Browser2021_1/schema/record/salesorder.html

Searches are a little complicated because they can have multiple pages of results and you have to juggle some things but this might give you a starting point
```php
        $tmp = new TransactionSearchBasic();
        $tmp->dateCreated = new SearchDateField();
        $tmp->dateCreated->operator = SearchDateFieldOperator::onOrAfter;
        $tmp->dateCreated->searchValue = date(DATE_ISO8601, $watermark);
        $search = new SearchRequest();
        $search->searchRecord = $tmp;
        $result = $client->search($search);
        // if there are more results continue paging through search with something like SearchMoreWithId using $result->searchId.
```
